### PR TITLE
chore: update docs

### PR DESCRIPTION
**Pull request by salmanm**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,6 +167,15 @@ fastify.route({
...
 
 
+## Securing the secret
+
+The `secret` shown in above code is strictly just a sample. In all cases, you'd need to make sure that the `secret`
+- **Never** hard-coded in the code or `.env` files or anywhere in the repository
+- Stored in some external services like KMS, Vault or something similar
+- Read at run-time and supplied in this option
+- Is long enough
+- Is truly random sequence of characters
```
Comment (simoneb) [inline README.md:177]: be consistent with the wording. either begin the sentence in each point with "Is" or don't. I would not do it and end the introductory sentence with "is" instead:

> that the `secret` is:
> - ...
> - ...

File: README.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ fastify.register(require('fastify-cookie'))
 fastify.register(require('fastify-csrf'))
 
 // if you want to sign cookies:
-fastify.register(require('fastify-cookie'), { secret: 'supersecret' })
+fastify.register(require('fastify-cookie'), { secret: 'supersecret' }) // See following section to ensure security
```
Comment (mcollina) [inline README.md:74]:
```suggestion
fastify.register(require('fastify-cookie'), { secret }) // See following section to ensure security
```

let's make it not working as a copy-paste, so the user will have to edit.

File: README.md [hunk 3]
```diff
@@ -167,6 +167,17 @@ fastify.route({
...
 
 
+## Securing the secret
+
+The `secret` shown in above code is strictly just a sample. In all cases, you'd need to make sure that the `secret` is:
+- **Never** hard-coded in the code or `.env` files or anywhere in the repository
+- Stored in some external services like KMS, Vault or something similar
+- Read at run-time and supplied in this option
+- Long enough
+- Truly random sequence of characters
```
Comment (mcollina) [inline README.md:177]: Is there a module or service we can recommend to generate this?

File: README.md [hunk 4]
```diff
@@ -167,6 +167,17 @@ fastify.route({
...
+
+Apart from these safeguards, it's extremely important to use HTTPS for your website/app to avoid a bunch of other potential security issues like [MITM](https://en.wikipedia.org/wiki/Man-in-the-middle_attack) etc.
```
Comment (mcollina) [inline README.md:179]: Maybe add a link to letsencrypt

**Review by mcollina — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (salmanm) [reply README.md]: Added hyperid & crypto-random-string

Comment (simoneb) [reply README.md]: I don't think hyperid can be considered a truly random char sequence generator in the sense we intend here. It does generate unique ids, just not random

Comment (salmanm) [reply README.md]: I just thought a generated hyperid string is good enough for a cookie secret, but yes, technically you're right.

Comment (simoneb) [reply README.md]: In this context it isn't. If you can guess what id is generated and you get hold of an encrypted cookie, you can decrypt it. In this context we need cryptographically secure secrets.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**